File: application/src/main/typescript/packages/react-simplicity/src/domain/descriptors/CollectionDescriptor.ts
import NodeDescriptor from "./NodeDescriptor";
import Basic from "../../mapper/annotations/Basic";
import Entity from "../../mapper/annotations/Entity";
import ObjectDescriptor from "./ObjectDescriptor";

@Entity("CollectionDescriptor")
export default class CollectionDescriptor extends NodeDescriptor    {

    $type = "CollectionDescriptor"

    @Basic()
    items : ObjectDescriptor

}

File: library/json-mapper2/src/main/scala/com/anjunar/scala/schema/model/CollectionDescriptor.scala
package com.anjunar.scala.schema.model

import com.anjunar.scala.mapper.annotations.{PropertyDescriptor, IgnoreFilter}

import scala.beans.BeanProperty
import scala.compiletime.uninitialized

@IgnoreFilter
class CollectionDescriptor extends NodeDescriptor {
  
  @PropertyDescriptor(title = "Items")
  var items : NodeDescriptor = uninitialized
  
}
